packages/client/src/app/jobs/executeJobModal.tsx

```diff
@@ -12,6 +12,10 @@ interface ExecuteJobModalProps {
 
 export const ExecuteJobModal = ({ closeModal, devices, jobId }: ExecuteJobModalProps): JSX.Element => {
   const [selectedDevices, setSelectedDevices] = useState<Selection>();
+  const [filter, setFilter] = useState('');
+
+const filteredDevices = devices?.filter(device => 
+  device.deviceId?.includes(filter) || device.origin?.includes(filter)) || [];
```

Comment:
You should memoize the filtering by using the useMemo hook. Also, as mentionned by GitHub, your indentation isn't right here.